server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java

```diff
@@ -86,16 +81,11 @@ public class FileSettingsService extends AbstractLifecycleComponent implements C
      * @param stateService an instance of the immutable cluster state controller, so we can perform the cluster state changes
      * @param environment we need the environment to pull the location of the config and operator directories
      */
-    public FileSettingsService(
-        ClusterService clusterService,
-        ReservedClusterStateService stateService,
-        Environment environment,
-        NodeClient nodeClient
-    ) {
+    public FileSettingsService(ClusterService clusterService, ReservedClusterStateService stateService, Environment environment) {
         this.clusterService = clusterService;
         this.stateService = stateService;
         this.operatorSettingsDir = environment.configFile().toAbsolutePath().resolve(OPERATOR_DIRECTORY);
-        this.nodeClient = nodeClient;
+        this.eventListeners = new ArrayList<>();
```

Comment:
I think this is properly synchronized as written here, but there is a risk that a future reshuffle might break that. Can we use something stronger just to be sure? (a `CopyOnWriteArrayList` is good enough I think).